packages/core/lib/segments/attributes/remote_request_data.js

```diff
@@ -7,15 +7,17 @@ var { stripQueryStringFromPath } = require('../../utils');
  * @param {http.ClientRequest|https.ClientRequest} req - The request object from the HTTP/HTTPS call.
  * @param {http.IncomingMessage|https.IncomingMessage} res - The response object from the HTTP/HTTPS call.
  * @param {boolean} downstreamXRayEnabled - when true, adds a "traced": true hint to generated subsegments such that the AWS X-Ray service expects a corresponding segment from the downstream service.
+ * @param {object} [options] - Optional HTTP Request options, which will take precedence over header data to populate request metadata.
  */
 
-function RemoteRequestData(req, res, downstreamXRayEnabled) {
-  this.init(req, res, downstreamXRayEnabled);
+function RemoteRequestData(req, res, downstreamXRayEnabled, options) {
+  this.init(req, res, downstreamXRayEnabled, options);
 }
 
-RemoteRequestData.prototype.init = function init(req, res, downstreamXRayEnabled) {
+RemoteRequestData.prototype.init = function init(req, res, downstreamXRayEnabled, options) {
+  const useOptionsHost = options && options.hostname;
```

Comment:
Isn't the original issue wanting to read from the request url, not a fixed option?

Reply:
The original issue said it was problematic to read from the header, and instead preferred reading from the `options` object. But I guess reading from the `request` host would work just as well and not require a signature change.

Reply:
Yeah let's just change the logic to read the request URL. It seems like a bug we use the host header (should suspect other languages / instrumentations aren't doing so) so just a simple be bugfix

Reply:
Updated to prefer `request.host` and fall back to the header value. It looks like `request.host` was only [added in Node 12](https://nodejs.org/api/http.html#http_request_host), so I guess that's why it was this way before.